packages/compartment-mapper/src/node-modules.js

```diff
@@ -344,14 +346,18 @@ const graphPackage = async (
     devDependencies = {},
   } = packageDescriptor;
   const allDependencies = {};
-  assign(allDependencies, commonDependencyDescriptors);
-  for (const [name, { spec }] of Object.entries(commonDependencyDescriptors)) {
-    allDependencies[name] = spec;
+  for (const [name, descriptor] of Object.entries(
+    commonDependencyDescriptors,
+  )) {
+    if (Object(descriptor) === descriptor) {
```

Comment:
OK, `Object(x) === x` is new to me. what does it do?

Reply:
This is the terse way of checking `typeof x === 'object' && x !== null`. `Object(x)` returns `x` iff `x` is an object and not null, otherwise boxes primitives.